src/domain/alerts/alerts.repository.ts

```diff
@@ -142,4 +165,33 @@ export class AlertsRepository implements IAlertsRepository {
       });
     }
   }
+
+  private async _notifySafeSetup(args: {
+    chainId: string;
+    newSafeState: Safe;
+  }): Promise<void> {
+    const emails = await this.emailRepository.getVerifiedEmailsBySafeAddress({
+      chainId: args.chainId,
+      safeAddress: args.newSafeState.address,
+    });
+
+    if (!emails.length) {
```

Comment:
I think we should have this condition as soon as we know the safe address. If there are no verified email addresses we shouldn't need to do all the mapping.

Reply:
I agree, good catch. It should be a common precondition before emitting any type of alert.

Reply:
I would suggest we keep this optimisation in mind and include it once we agree upon the [retrieval of the Safe address for unknown transactions](https://github.com/safe-global/safe-client-gateway/pull/923).

Reply:
I've included the suggestion in d8d3ab5081b8c96368aeebab93455741f2210c6e with a comment as this will need to be revisited when we are in agreement regarding the [retrieval of the Safe address for unknown transactions](https://github.com/safe-global/safe-client-gateway/pull/923).